### PR TITLE
Zombienet re-enable upgrade test

### PR DESCRIPTION
**Pull request by pepoviola**
Re-enable the upgrade test, that was disable by #4873.
CC @ordian.

### CODE REVIEW

**Review by ordian — APPROVED** [review]
Thanks!

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -682,6 +682,35 @@ zombienet-tests-parachains-disputes:
   tags:
     - zombienet-polkadot-integration-test
 
+zombienet-test-parachains-upgrade-smoke-test:
+  stage:                           deploy
```
Comment (alvicsam) [inline .gitlab-ci.yml:686]:
```suggestion
  stage:                           stage4
```
Stages were renamed and became artificial in order to use DAG in the pipeline.

**Review by alvicsam — CHANGES_REQUESTED** [review]

File: .gitlab-ci.yml (hunk 1 above)
Comment (ordian) [reply .gitlab-ci.yml:686]: why stage4? Shouldn't it be the same stage as other zombienet tests, which is stage3?

Comment (alvicsam) [reply .gitlab-ci.yml:686]: fully agree, fixed suggestion

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -682,6 +682,35 @@ zombienet-tests-parachains-disputes:
...
 
+zombienet-test-parachains-upgrade-smoke-test:
+  stage:                           stage3
+  image:                           "${ZOMBIENET_IMAGE}"
+  <<:                              *kubernetes-env
+  <<:                              *zombienet-refs
+  needs:
+    - job:                         publish-polkadot-debug-image
+    - job:                         publish-malus-image
+    - job:                         publish-test-collator-image
```
Comment (alvicsam) [inline .gitlab-ci.yml:693]:
```suggestion
    - job:                         publish-test-collators-image
```

**Review by alvicsam — CHANGES_REQUESTED** [review]

**Comment by sandreim** [comment]
bot merge

**Review by radupopa2010 — APPROVED** [review]

**Comment by sandreim** [comment]
bot merge

**Pull request merged**